src/Solver.cpp

```diff
@@ -489,52 +655,212 @@ void Solver::initializeCmfd(){
-  Cell* cell;
+  log_printf(NORMAL, "Computing the flux...");
 
-  /* Set each tally to zero to begin with */
-  #pragma omp parallel for
-  for (int r=0; r < _num_FSRs; r++)
-    FSR_segment_tallies[r] = 0;
+  /* Clear all timing data from a previous simulation run */
+  clearTimerSplits();
 
-  /* Iterate over all azimuthal angles, all tracks, and all Track segments
-   * and tally each segment in the corresponding FSR */
-  #pragma omp parallel for private (num_segments, curr_segment)
-  for (int i=0; i < _tot_num_tracks; i++) {
+  /* Start the timer to record the total time to converge the flux */
+  _timer->startTimer();
 
-    num_segments = _tracks[i]->getNumSegments();
-    segments = _tracks[i]->getSegments();
+  FP_PRECISION residual;
 
-    for (int s=0; s < num_segments; s++) {
-      curr_segment = &segments[s];
-      FSR_segment_tallies[curr_segment->_region_id]++;
+  /* Initialize data structures */
+  initializePolarQuadrature();
+  initializeExpEvaluator();
+
+  /* Initialize new flux arrays if a) the user requested the use of 
+   * only fixed sources or b) no previous simulation was performed which
+   * initialized and computed the flux (e.g., an eigenvalue calculation) */
+  if (only_fixed_source || _num_iterations == 0) {
+    initializeFluxArrays();
+    flattenFSRFluxes(0.0);
+  }
+
+  initializeSourceArrays();
+  initializeFSRs();
+  zeroTrackFluxes();
+
+  /* Compute the sum of fixed, total and scattering sources */
+  computeFSRSources();
+
+  /* Source iteration loop */
+  for (int i=0; i < max_iters; i++) {
+
+    transportSweep();
+    addSourceToScalarFlux();
+    residual = computeResidual(SCALAR_FLUX);
+    storeFSRFluxes();
+
+    log_printf(NORMAL, "Iteration %d:\tres = %1.3E", i, residual);
+
+    /* Check for convergence of the fission source distribution */
+    if (i > 1 && residual < _converge_thresh) {
+      _num_iterations = i;
+      _timer->stopTimer();
+      _timer->recordSplit("Total time");
+      return;
     }
   }
 
-  /* Loop over all FSRs and if one FSR does not have tracks in it, print
-   * error message to the screen and exit program */
-  #pragma omp parallel for private (cell)
-  for (int r=0; r < _num_FSRs; r++) {
+  log_printf(WARNING, "Unable to converge the flux");
+
+  _num_iterations = max_iters;
+  _timer->stopTimer();
+  _timer->recordSplit("Total time");
+}
+
 
-    if (FSR_segment_tallies[r] == 0) {
-      log_printf(ERROR, "No tracks were tallied inside FSR id = %d. Please "
-                 "reduce your track spacing, increase the number of "
-                 "azimuthal angles, or increase the size of the FSRs", r);
+/**
+ * @brief Computes the total source distribution by performing a series of 
+ *        transport sweep and source updates.
+ * @details This is the main method exposed to the user through the Python
+ *          interface to compute the source distribution, e.g., for a fixed
+ *          and/or external source calculation. This routine makes an initial 
```

Comment:
I'm fairly indifferent to the name choice, but if I had to pick I would vote for `solve...` instead of `compute...`.